lib/fb/post.rb

```diff
@@ -2,25 +2,43 @@
 # @see http://www.rubydoc.info/gems/Fb/
 module Fb
   # Fb::Post reprensents a Facebook post. Post provides getters for:
-  #   :id, :title, :url, :created_time, and :type.
+  #   :id, :url, :created_at, :type, :message, and :length.
   class Post
+
     # @option [String] the post’s unique ID.
     attr_reader :id
 
-    # @option [String] the post’s type.
-    attr_reader :type
+    # @option [String] the post’s permanent url.
+    attr_reader :url
 
     # @option [Time] the post’s creation time.
     attr_reader :created_at
 
+    # @option [String] the post’s type.
+    attr_reader :type
+
+    # @option [String] the post’s message.
+    attr_reader :message
+
+    # @option [String] the attached video's length if type video.
+    attr_reader :length
+
     # @param [Hash] options the options to initialize an instance of Fb::Post.
-    # @option [String] :id the post id.
-    # @option [String] :type the post’s type.
-    # @option [String] :created_time the post’s creation time in iso8601 format.
+    # @option [String] :id The post id.
+    # @option [String] :message The status message in the post or post story.
+    # @option [String] :url URL to the permalink page of the post.
+    # @option [String] :created_time The time the post was initially published.
+    # @option [String] :type A string indicating the object type of this post.
+    # @option [String] :length The length of the attached video.
     def initialize(options = {})
       @id = options[:id]
+      @url = options[:permalink_url]
+      @created_at = Time.strptime(options[:created_time], '%Y-%m-%dT%H:%M:%S+0000')
       @type = options[:type]
-      @created_at = Time.parse(options[:created_time]) if options[:created_time]
+      @message = options[:message]
+      @length = options.fetch(:properties, []).find(-> { {'text' => 'n/a'} }) do |p|
+        p['name'] == 'Length'
```

Comment:
@S1v4 You can use `property` rather than `p` since `p` would overwrite the `puts` method of Ruby